File: MealPlanner/src/main/java/com/nico/mp/domain/Recipe.java
package com.nico.mp.domain;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Getter
@Setter
@Entity
public class Recipe {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private Long userId;
	
	private String name;
	
	private Integer servings;
	
	private Integer prepTime;
	
	private String notes;

	@OneToMany
	private List<Ingredient> ingredients;

	@OneToMany
	private List<Instruction> instructions;

}

File: MealPlannerApp/src/app/models/recipe.ts
import { Ingredient } from "./ingredient"
import { Instruction } from "./instruction"

export type Recipe = {
    id: number,
    userId: number,
    name: string,
    servings: number,
    prepTime: number,
    notes: string,
    ingredients: Ingredient[],
    instructions: Instruction[]
}